Some upgrades

im-coder-lg opened the pull request:
- [ ] Need to brush up on the website now.

https://github.com/Futura-Py/futura-py.github.io/pull/2 is the source PR where I get the updates. I know this just got mentioned, and this is what I'm trying to do.

Moosems commented:
Is it just me or did every single test fail 😄?

im-coder-lg commented:
It just did lol

I added each page in HTML. So there are issues with linking and accessing files. They become HTML when built, but not right now. I must fix this later.

im-coder-lg commented:
I tried fixing it, but it was too late here. So I just left it. Any changes I made are on my Gitpod workspace, and I can carry on from where I left.

im-coder-lg commented:
Seems like it kinda worked. The page looks like the one time one of my websites got its code corrupted.

not-nef commented:
the page looks like the css doesnt work

not-nef commented:
but can we keep deploying with gh pages?

sumeshir26 commented:
> but can we keep deploying with gh pages?

Agreed, GH Pages has that simplicity that Netlify does not

im-coder-lg commented:
But we wouldn't have our dearest deploy previews. Why not both? We could use Vercel than Netlify. I just have to configure the variables to tune it to Astro. Meanwhile, you guys must work on merging Futura-Py/futura-py.github.io#2 so that I can add those changes here.

im-coder-lg commented:
Maybe from here, I have to use modules like `<Card />`. I'll try and tell here.

not-nef commented:
ok, so can you one last time list the benefits of astro over bare bones html, and why do we need netlify or vercel at all? the previews are not that important because if you want to change something you will be on a proper machine anyway. and if its really important we can quickly switch the deployment branch.

im-coder-lg commented:
> ok, so can you one last time list the benefits of astro over bare bones html, and why do we need netlify or vercel at all? the previews are not that important because if you want to change something you will be on a proper machine anyway. and if its really important we can quickly switch the deployment branch.

JS modules, easily controlled, just one command to start the website, CMSs can be used if wanted, better theming(tailwind CSS+Astro Community Components).

Netlify and Vercel are mainly compatible with Astro. No matter what you do, GH Pages won't deploy previews. Now Vercel is on par with Netlify with the deploy preview collaboration features.

Yeah, you would be on a machine, so you would see the change. Others can witness the live change with previews. 
It's generally a nice addition to the codebase. If your website isn't deployed, you will know why. There are zero liits on deployments, but there is on bandwidth(looking right at Netlify).

Switching the deployment branch is just sort of repeated, and would be tedious. Instead, keeping it like this, you know what will happen, and you won't have to break your head over it. If you set all sidejobs up, the goal would be clear, and everything will help you. Something I tend to follow subconsciously, foresight. And anyway, using a site generator is a good choice in terms of contributions. Maybe you wouldn't have Node.js, but you can make a Gitpod account for free, and that has 50 hours per month for free, with paid boosts. I use that to make my changes here.

im-coder-lg commented:
And I don't kind of understand this stereotype that if you load something up, it is hard to manage. It is true 90% of the time. The rest of the time, it is managed. Astro uses this island thingy, it loads up what the user needs to see, rather than everything at once. At build time, everything becomes HTML, and public assets stay as that. JS components are used with HTML code here, so there isn't much to trouble learning with, except the values available. Trust me, try it. Use Gitpod, it is snappy when you need it to be. It ships with LTS versions of NodeJS and NPM, so you can test it freely when you have time. I have my website, https://imcoderlg-rkt.netlify.app, and this is the PageSpeed result for desktop:

![Screenshot from 2023-09-28 22-28-20](https://github.com/im-coder-lg/futura-astrosite/assets/76845820/a33a039e-6e0f-4073-9b32-9b0702a86009)

I am not that popular, since SEO and accessibility isn't at best right now. I'm not searching for fame, btu that is the website's performance! It is very snappy too on experience.

Check out our website:
![Screenshot from 2023-09-28 22-33-57](https://github.com/im-coder-lg/futura-astrosite/assets/76845820/8aa3896e-09fe-413d-a71f-57596148bfef)
![Screenshot from 2023-09-28 22-33-54](https://github.com/im-coder-lg/futura-astrosite/assets/76845820/e1f27244-4476-4603-9da3-37fc91d688ee)

100% performance speed. It is basically HTML right now, and that just proves what I said. It loads what you need, and discards the useless stuff. It might reduce later on, but that is very good for starters. It just proves that in terms of Astro, speed and repo size are not proportional to each other. They are different, as it should be with Astro.

And yeah, I'm thinking of testing both Netlify and Vercel. Could be worth it, if we know which one is the right choice.

im-coder-lg commented:
I used the Vercel deployment, to get some insight on the CDN we could use for testing purposes.

not-nef commented:
i dont really know about all of this. i really want to keep everything as it is. i dont need all of the js modules and deployment previews since most of the time i will push my changes to main anyway. i think all of this just clutters the repo. i mean look at the job list. 7 jobs?? and all the comments from the bots. and with gh pages we can keep our nice futura-py.github.io url. i will let the other guys decide it but i want to let you know that personally im against it. the site performance is absolutely fine.

im-coder-lg commented:
Well, we can remove Netlify if you want. Vercel seems to work well.

im-coder-lg commented:
And 1 job isn't really a job; it's the WIP app, to prevent merging by accident. Drafting does that, but I keep it on my repos, as a personal app.
Netlify has only 1 check, the rest are diff updates. Vercel has 1, and the other one is a feedback resolve update.

not-nef commented:
i want to keep it as it is. @Moosems what do you think?

sumeshir26 commented:
> i dont really know about all of this. i really want to keep everything as it is. i dont need all of the js modules and deployment previews since most of the time i will push my changes to main anyway. i think all of this just clutters the repo. i mean look at the job list. 7 jobs?? and all the comments from the bots. and with gh pages we can keep our nice futura-py.github.io url. i will let the other guys decide it but i want to let you know that personally im against it. the site performance is absolutely fine.

Completely agree, this is such a tiny website and we are just overcomplicating the code. And the fututa-py.github.io site is nice too. The deploy previews are completely useless imo because even if you use gitpod you can see the preview there anyway.

im-coder-lg commented:
Honestly, I don't get you guys. It is not complex at all. I mean, Astro? 
- Very fast
- Has a configuration to make everything easier
- Supports HTML, JS, and TS out of the box. Will support React, Vue, Markdown, Markdown + JSX with integrations

`.astro` files use HTML everywhere. The front-matter part has the JS imports for Astro components, which are used as HTML only. Everything here is clearer than ever, and I want you guys to try it. Seriously. I mean, what is the use of my ideas if I can't work on it? It happened in Weather too, I couldn't figure out head or tail with @Moosems's OOP changes. No offence here, it was good, and it works, but I don't really know OOP. I tried, but it didn't work, and I stuck to traditional Python for me. It is hard to figure out what goes where, and trust me, the original code was from my Google Colab, where I did it in the simplest way possible. Now, I don't get how everything now works.
Look, it happened there, because I didn't get it. But here, I tried using the simplest things. I added HTML _only_ for the pages, and used the same CSS from the PR, adding only one thing, a "homepage" component that needs to be renamed since Astro couldn't render the homepage icon then, now making it configurable and controllable easily. I even have uncommitted changes on my Gitpod, for the card view for our projects. I am trying to convince you guys to use a CDN to be even clearer. Netlify is free and has fewer limitations than Vercel, shows the diff update with those "blank" checks, and even has an updated comment for visiting the deploy preview. It came there multiple times, since I deleted it once, and added it back. I am trying hard so that you guys can see the potential here, even trying to add the card view and maybe a more easily customisable navbar, adding icons, and useful components like the "breadcrumbs" one for the projects, and I am trying to find out how Tailwind CSS can be used here. Look, don't take me as angry, but I just want myself to be more clear and understood since I always am not. I do that to myself. I want the website to be easier to add stuff to, and I don't want it to kind of get redundant over time. It must be one of the best and fastest, hence this kind of complexity.
I read this quote from a Steve Jobs biography. Paraphrasing it here, Tim Cook, after Jobs's passing, tells a lesson he learnt from the legend. That simplicity is the ultimate sophistication, but achieving it requires complexity, till a level when the end product is simple. I believe that, hence this might look complex. But try it. It is always simple when you understand, and since this is a near-replica of the actual site, I know you will.

I reckon this website one day will be everything but small, and that is without Astro, and as barebones HTML.

Moosems commented:
I believe you misunderstand a few things:
- We don't dislike your ideas, we simply don't see the overall objective. We don't see this becoming very big and right now we don't even have the base pages, let alone making it part of a large JS framework!
- The Weather app using OOP: dude for all your talk of complexity and whatnot you have no room to talk, not only that, it was messy and while you may have been able to read it, it was a pain to decipher everything. If you take the time to learn classes and basic OOP and actual standards for all your talk of using all the norms like a src folder and whatnot it would make much more sense.

Moosems commented:
Plus, when it comes to previews, I can just run the html file in my browser and it works perfectly!

not-nef commented:
> Honestly, I don't get you guys. It is not complex at all. I mean, Astro?
> 
> * Very fast
> * Has a configuration to make everything easier
> * Supports HTML, JS, and TS out of the box. Will support React, Vue, Markdown, Markdown + JSX with integrations

This would be convincing if our site was not fast and would use React, Vue, Markdown, Markdown + JSX with integrations.

> I couldn't figure out head or tail with @Moosems's OOP changes. No offence here, it was good, and it works, but I don't really know OOP. I tried, but it didn't work, and I stuck to traditional Python for me.

Not related but its kind of funny how this was basically me before you guys bullied me into using it and now i love it.

Moosems commented:
And @not-nef wasn't it worth it?

not-nef commented:
IT WAS SO WORTH IT! the codebase is like 20% of its original size now and it all works so beautifully and simple i thank you guys so much for it!!

Moosems commented:
@im-coder-lg Please don't take it wrong, I think this would be something great to do in the future that the site needs some base stuff before we add a framework. Let's please focus on the foundation before building the house.

not-nef commented:
@im-coder-lg i really dont want you to feel like we dont want to implement this because we think this idea is bad overall. we just dont even really use javascript yet! the only js we use is for the github api request and building the member list and thats it! theres really not much that can be sped up or simplified just yet. if this project gets huge one day, we can still get back to this and see if the site would benefit by adding such a framework, but currently we just dont.

sumeshir26 commented:
We will for sure use Astro if we ever get to a more complicated website, since this is basically just a few simple HTML pages linked together, nothing more than that. Im so sorry your work went to waste :(

im-coder-lg commented:
You guys do that. Build the base. I told you this before, I tried OOP, but it didn't work. Mind you, that was from an online course for my father. I just sat next to him and coded as they said. Everything except OOP worked, so I dropped it.
I can't work with you guys on the website if you guys just keep adding your changes in your way. Alright? I don't want to clone a website repeatedly on my computer and check if the website works. I don't want to switch deployment branches just to see if the website works.
You guys are right. My work was wasted. But then, you keep adding things in such a way that I can't make my additions. Even if the original Weather code was complex, telling that up straight might have saved us a few minutes. Okay? I didn't voice out my opinions, and I just did. You guys did the same, but not sooner for me to understand your point of view.
I'll tell you this: if you guys bullied even me to try OOP, I might have done so. I had so much time before, and that was wasted. That might have been my mistake. Okay?
If you want me to stop, just say it. I want to make the website better in such a way all of us can use it. But in the end, it is your choice. In the end, I feel like I can't decide anymore.
If you tell me to stop, I will do so. I will however commit my final changes, and make this repo private or host it on GitLab, something I prefer more, by next week. I will try working on this to make it simpler. I believe when the time comes, this Astrosite will be ready.
Thanks for voicing your opinions. Though they have hurt me in some ways, more in some places than others, you guys gave me the roadmap I need. I just hope it would have been more gentle. I know I am not like that. I like to be direct. But I believe this situation just needed some indirect help, though this one was a mix of both.

not-nef commented:
:(      i didnt want to hurt you, i just thought the way we are doing things currently is the simplest and the one to go

im-coder-lg commented:
> :( i didnt want to hurt you, i just thought the way we are doing things currently is the simplest and the one to go

It's fine. I can tolerate it. Life isn't without pain, and pain is needed when we must change for the better. But I must know if I should stop. I added the final commit, the last changes I have made. Just tell me to stop if I must. I need to know.

im-coder-lg commented:
I think it is time, that if this project/test/attempt were to be halted, this PR must be merged. I am just trying to keep all loose ends tied up. That way, when I finally have time to think of something after the foundation is established, I will know what to do and what _not_ to do. I wouldn't have to focus on this.
